### PR TITLE
Tape: Add node events emitted by interface Test

### PR DESCRIPTION
**Pull request by janopae**
You can actually use the function `on` in node on every object that `emit` is called upon. Tape calls `emit` on itself quite a few times, you can find all those events by searching for `emit(` in the Tape source code.

I stumbled upon this porting a Tape test to TypeScript that depended on the test.on('end') event, in order to do a teardown after each test.

Unfortunately, I didn't understand exactly for all of these events at which stages they are emitted, that's why I documented none of them. But I could find out the argument types the events can possibly be emitted with, which alone was a whole lot of work.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>

### CODE REVIEW

File: types/tape/index.d.ts [hunk 1]
```diff
@@ -238,5 +238,23 @@ declare namespace tape {
          * Assert that string does not match the RegExp regexp. Will throw (not just fail) when the first two arguments are the wrong type.
          */
         doesNotMatch(actual: string, expected: RegExp, msg?: string, extra?: AssertOptions): void;
+        
+        /**
+         * Tape emits some node events you can hook up to.
+         */
+        on(event: 'prerun' | 'run'  | 'end', callback: () => any );
```
Comment (ljharb) [inline types/tape/index.d.ts:245]: all these callbacks should be void; not returning any

**Review by ljharb — CHANGES_REQUESTED** [review]
This isn’t including “once” - due to https://github.com/substack/tape/blob/cd81a404d7a8308b9aad50d2aa2c92e330351d1e/lib/test.js#L28 , the entire EventEmitter interface is present here - does DT have a pattern for applying that?

**Comment by janopae** [comment]
> This isn’t including “once” - due to https://github.com/substack/tape/blob/cd81a404d7a8308b9aad50d2aa2c92e330351d1e/lib/test.js#L28 , the entire EventEmitter interface is present here - does DT have a pattern for applying that?

I don't know. For ws, they added all the methods to the interface:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43170

But I prefer not supporting `emit`, `addListener`, `off` and `removeListener` here, as they make not so much sense for Tape.

**Comment by ljharb** [comment]
`off` and `removeListener` make sense on anything that has an `on`/`addListener`; I think those need to be added too.

I agree on `emit`, but what i'd rather see here is adding EventEmitter the proper way, and then *overriding* the type of `emit` so that it's not available in the tape interface.

**Comment by ljharb** [comment]
um, closed after only 7 days of inactivity from the author?

cc @sandersn can this be reopened, and the bot dialed down a bit to be a bit less hostile?

**Comment by sandersn** [comment]
@orta was the one tuning the bot recently. Can you take a look @orta?

**Comment by orta** [comment]
:wave: this one got a bit unlucky because it was during a period when we weren't poking folks before they became inactive. This is the current workflow for closing a PR (I'm working on bot docs ATM):

---

### Idle PR Removal

When a PR:

 - Has merge conflicts, Travis CI is failing or has Reviews which reject the most recent commit
 - Has not had any commits/comments/reviews/review comments in the last 5 days
 
It will get a ping that it has two days for something to happen.

Then, assuming no new activities, a PR which:

- Has merge conflicts, Travis CI is failing or has Reviews which reject the most recent commit
- Has not had any commits/comments/reviews/review comments in the last 7 days

Will be closed.

--- 

Which I think is pretty reasonable - normally we ping folks and that keeps things alive anyway, I'll re-open which gives it another

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## tape ([<kbd>unpkg</kbd>](https://unpkg.com/browse/tape@latest/))
was missing the following properties:
1. getHarness
1. Test
1. test
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7ee783bf07dc142d433a7ded535a22b3c29281ec
</p>

**Comment by ljharb** [comment]
@orta people have busy lives; 7 days seems way too short to me. I'd suggest a month at least.

**Comment by orta** [comment]
👍 aye, we'll bump it to a month

**Pull request closed without merging**